### PR TITLE
Add support for adding custom elements

### PR DESCRIPTION
**Pull request by mikabytes**
Based on #550, this PR indeed adds support for importing files that define custom elements. It has a few quirks though.

### Quirks 
- I want it to be treated as a named export, but JavaScript doesn't allow me to specify such metadata. The consequence is that imports can be added, but not matched or removed.
- Correct me if I'm wrong, but it seems ImportJS doesn't have support for "that which has a name but is not a named export". My solution is horrible, adding it as a named export and then stripping it away as a side-effect later.
- I lacked the energy to write tests for this. I will, but first, I would like to invite your comments on what I've done so far.
- I'm not too happy about the way I'm checking the type-of-export based on the adding <> characters to the name. I AM happy about using <> to have ImportJS give me distinct messages about importing a custom element though.
- I didn't find a suitable way to traverse the AST to find all the statements, so I got lazy and used a regular expression instead. This means that it will incorrectly import files where this statement has been commented out.
- No configuration option to disable this feature (yet)

### One idea
Maybe we could add support for "that which has a name but is not named in JS spec" as a comment? Maybe something like:

```
import './my-custom-element.js' // <my-custom-element>
```

OR

```
import /* <my-custom-element> from */ './my-custom-element.js'
```

Separating multiple named exports with comma. Or something. I'm happy to hear your ideas.

### other
Not too bad for 3 hours work on a foreign codebase.... :)

### CODE REVIEW

**Comment by trotzig** [comment]
Apologies for the delay here! I hope to have a look at this later today.

File: lib/findExports.js [hunk 1]
```diff
@@ -404,6 +404,11 @@ function findRootNodes(ast) {
   return realRootNodes;
 }
 
+function findUnnamedExports(data) {
+  const regex = /customElements\.define\s*\(\s*['"]([0-9a-z-]+)['"]\s*,\s*[a-zA-Z0-9_$]+\s*\)/g
+  return Array.from(data.matchAll(regex)).map(t => `<${t[1]}>`)
+}
+
```
Comment (trotzig) [inline lib/findExports.js]: The regex here is fine, but make sure to add tests that show the behavior in different edge-cases so that they're clearly documented (like the commented-out code case you mentioned).

**Comment by trotzig** [comment]
Excellent work!

> I want it to be treated as a named export, but JavaScript doesn't allow me to specify such metadata. The consequence is that imports can be added, but not matched or removed.

You're right that ImportJS doesn't have support for that, but you're wrong about your solution being horrible. :) 

> I'm not too happy about the way I'm checking the type-of-export based on the adding <> characters to the name. I AM happy about using <> to have ImportJS give me distinct messages about importing a custom element though.

I think this is fine. 


> `import /* <my-custom-element> from */ './my-custom-element.js'`

I like this idea. If you can make this work, go for it. If not, I think it's fine if removing the imports is a manual process.

File: lib/visitIdentifierNodes.js [hunk 2]
```diff
@@ -102,10 +116,9 @@ export default function visitIdentifierNodes(
       queue.unshift(...itemsToAdd);
       continue; // eslint-disable-line no-continue
     }
-    const normalizedNode = normalizeNode(current.node, current.context);
-    if (normalizedNode) {
+    normalizeNode(current.node, current.context, (normalizedNode) => {
       visitor(normalizedNode, current.context);
```
[the remark this answers is not in the document]
Comment (mikabytes) [reply lib/visitIdentifierNodes.js]: > optionally returning an array
Having mixed return values have a bit of a code smell for me, having to test the object in order to use it. This way the function is more honest: "I'm going to produce zero, one or several normalizedNodes".

Of course, we could always return an array. Not sure why I went with the callback pattern. I don't like it, I never liked it :) An iterable of some kind makes the most sense. I'll convert it to a generator if you agree this would be a most pleasant way forward?

File: lib/findExports.js (hunk 1 above)
Comment (mikabytes) [reply lib/findExports.js]: I gave this some more thought and realized the problems don't end at commented code. It would also pick up the pattern inside of strings etc.

I'm becoming more certain that traversing the AST tree is the proper way to go. Do you have any indication on how much work that would involve? Guess it couldn't be too difficult?

**Comment by mikabytes** [comment]
> but you're wrong about your solution being horrible

Haha, agree to disagree ;-)

> import /* <my-custom-element> from */ './my-custom-element.js'

I'll have something ready once I have some time to do it. Might not be this week, we'll see!

File: lib/visitIdentifierNodes.js (hunk 2 above)
[the remark this answers is not in the document]
Comment (trotzig) [reply lib/visitIdentifierNodes.js]: I don't have a strong preference, you make the call :)

File: lib/findExports.js (hunk 1 above)
Comment (trotzig) [reply lib/findExports.js]: Hard to say, these things tend to be very explorative in nature. I usually TDD things like this, start with adding a test case to `findExports-test.js` and iterate from there. 

The AST traversing helpers in this library aren't super intuitive, and there might be better alternatives out there. But with a TDD approach things tend to work quite well.
